src/vm/index.md

```diff
@@ -118,14 +141,16 @@ In addition, during predicate mode if `$pc` is set to a value greater than the e
 
 A predicate that halts without returning Boolean `true` does not pass verification, making the entire transaction invalid. Note that predicate validity is monotonic with respect to time (i.e. if a predicate evaluates to `true` then it will always evaluate to `true` in the future).
```

Comment:
> During predicate mode, hitting any of the following instructions causes predicate verification to halt, returning Boolean `false`:
> 1. Any [contract instruction](./instruction_set.md#contract-instructions).
> 1. [JMP](./instruction_set.md#jmp-jump), [JI](./instruction_set.md#ji-jump-immediate), [JNE](./instruction_set.md#jne-jump-if-not-equal), [JNEI](./instruction_set.md#jnei-jump-if-not-equal-immediate), or [JNZI](./instruction_set.md#jnzi-jump-if-not-zero-immediate) with jump-to value less than or equal to `$pc` (these would allow loops). In other words, `$pc` must be strictly increasing.

The section above this needs to adjust the restriction on JMP to allow backwards jumps within the predicate range